catalog/dags/common/sensors/external_dag_sensor.py

```diff
@@ -0,0 +1,39 @@
+from airflow.sensors.external_task import ExternalTaskSensor
+from airflow.utils.state import State
+
+from common.constants import REFRESH_POKE_INTERVAL
+from common.sensors.utils import get_most_recent_dag_run
+
+
+class ExternalDAGSensor(ExternalTaskSensor):
```

Comment:
What's the reason to use a class that just extends init (and doesn't override any other behaviour) rather than a function that constructs `ExternalTaskSensor` and returns it directly, as in a factory function?

The only downside I can think of is that this creates an intermediary type with a slightly different name than the built-in Airflow one. That might be slightly more confusing to read stack traces for or keep track of `isinstance` usage (though that's probably unlikely to be used with this class, right?).

Reply:
Just a stylistic preference, but weakly held -- and frankly, doing it this way makes it harder to organize the code for reuse because of circular imports, so I'm definitely convinced to change it. Updated, thanks!